dds/DCPS/QOS_XML_Handler/XML_MemBuf_Intf.cpp

```diff
@@ -0,0 +1,235 @@
+#include "XML_MemBuf_Intf.h"
+#include "xercesc/util/XercesDefs.hpp"
+#include "xercesc/framework/MemBufInputSource.hpp"
+#include "xercesc/parsers/XercesDOMParser.hpp"
+#include "xercesc/sax/SAXParseException.hpp"
+
+#include "DataReaderQos_Handler.h"
+#include "DataWriterQos_Handler.h"
+#include "TopicQos_Handler.h"
+#include "PublisherQos_Handler.h"
+#include "SubscriberQos_Handler.h"
+#include "ParticipantQos_Handler.h"
+
+#include "dds/DCPS/debug.h"
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+XERCES_CPP_NAMESPACE_USE
+
+namespace OpenDDS {
+namespace DCPS {
+
+  QOS_XML_MemBuf_Handler::QOS_XML_MemBuf_Handler (void) :
+	  QOS_XML_Handler(),
+	  res_(new XML::XML_Schema_Resolver<XML::Environment_Resolver>()),
+	  eh_(new XML::XML_Error_Handler())
+  {
+	  try
+	  {
+		  // Call Initialize if not already called
+		  XMLPlatformUtils::Initialize();
+		  
+		  ////////////////
+    	  // Create parser
+		  parser_ =  new XercesDOMParser();
+
+		  // Create a DomImplementation
+		  DOMImplementation * domImpl = DOMImplementationRegistry::getDOMImplementation(XMLString::transcode("XML 1.0"));
+		  // Create a temporary document with a generic element
+		  // it shall be replaced in future with final version
+		  finalDoc_ = domImpl->createDocument(
+		  		XMLString::transcode(""),
+		  		XMLString::transcode("temp"),
+		  		NULL);
+
+	  } catch (const XMLException& toCatch) {
+
+	      char* message = XMLString::transcode(toCatch.getMessage());
+          ACE_ERROR ((LM_ERROR,
+          	ACE_TEXT ("QOS_XML_MemBuf_Handler::QOS_XML_MemBuf_Handler - ")
+          	ACE_TEXT ("Error during XML initialization! :\n<%C>\n"),
+          	message));
+	      XMLString::release(&message);
+	  }
+  }
+
+  QOS_XML_MemBuf_Handler::~QOS_XML_MemBuf_Handler (void)
+  {
+	  if (finalDoc_ != NULL)
+		  finalDoc_->release();
+	  if (parser_ != NULL)
+		  delete parser_;
```

Comment:
No need to check for null before calling delete. Also use 0 instead of NULL, NULL is not portable